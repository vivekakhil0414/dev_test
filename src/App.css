

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}



/* sidebar css */

.sidenav {
  height: 100%;
  width: 100px;
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  background-color: #3f0e40;
  overflow-x: hidden;
  padding-top: 20px;
  text-align: left;
}

.sidenav a {
  padding: 6px 8px 6px 5px;
  text-decoration: none;
  font-size: 15px;
  color: #ffffff;
  display: block;
  text-decoration: none !important;
}

.sidenav a:hover {
  color: #d0d0d0;
}

.main {
  margin-left: 100px !important; 
  padding: 30px 10px;
  max-height: calc(100vh - 164px);
  overflow-y: auto;
  width: calc(100% - 100px);
}

.footer {
  width: 100%;
  background-color: #1d0f1d;
  color: #fff;
  text-align: center;
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
}
.footer p{
  margin: 0px;
  padding: 20px 0px;
}
@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav a {font-size: 18px;}
}

/* sidebar css */