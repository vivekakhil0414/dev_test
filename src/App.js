import React, { useState, useEffect, Fragment } from "react";
import "./App.css";
import NavbarTop from "./components/Navbar/Navbar";
import ApolloClient from "apollo-boost";
import { ApolloProvider } from "react-apollo";
import Visitors from "./components/Visitors/Visitors";
import * as axios from "axios";
import Footer from "./components/footer/footer";
import { Pagination } from "react-bootstrap";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Home from "./components/Home/Home";

let url = "http://localhost:5000/"; // development

const client = new ApolloClient({
  uri: `${url}graphql`
});

let filterList = [
  { attr: "", val: "Filter" },
  { attr: "today", val: "Today" },
  { attr: "yesterday", val: "Yesterday" },
  { attr: "week", val: "Last Week" },
  { attr: "month", val: "This month" }
];
let limitList = [
  { attr: "", val: "Limit" },
  { attr: "5", val: "5" },
  { attr: "10", val: "10" },
  { attr: "50", val: "50" }
];

function App() {
  let [filter, setFilter] = useState("");
  let [active, setActive] = useState(1);
  let [limit, setLimit] = useState(10);

  useEffect(() => {
    axios
      .get("https://jsonip.com")
      .then(ip => {
        axios.get(`${url}?ip=${ip.data.ip}`).then(data => console.log);
      })
      .catch(e => console.log(e));
  }, [filter, limit]);

  const onChange = event => {
    setFilter(event.target.value);
  };

  const onChangeLimit = event => {
    setLimit(event.target.value);
  };

  function changePage(item) {
    setActive(item);
  }

  return (
    <ApolloProvider client={client}>
      <div className="App">
        <Router>
          <NavbarTop />
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/service" render={() => {
                return (
                  <Fragment>
                  <select className="selectpicker" onChange={onChange}>
                  {filterList.map(item => {
                    return (
                      <option key={item.val} value={item.attr}>
                        {item.val}
                      </option>
                    );
                  })}
                </select>
                <select className="selectpicker" onChange={onChangeLimit}>
                  {limitList.map(item => {
                    return (
                      <option key={item.val} value={item.attr}>
                        {item.val}
                      </option>
                    );
                  })}
                </select>
                <Visitors filter={filter} limit={limit} skip={active} />
                <Pagination
                  size="sm"
                  style={{
                    float: "right",
                    marginRight: "2em",
                    cursor: "pointer"
                  }}
                >
                  {[1, 2, 3, 4, 5].map(item => {
                    return (
                      <Pagination.Item
                        onClick={() => changePage(item)}
                        key={item}
                        active={item === active}
                      >
                        {item}
                      </Pagination.Item>
                    );
                  })}
                </Pagination>
                </Fragment>
                )
            }} />
          </Switch>
        </Router>
      </div>
      <Footer />
    </ApolloProvider>
  );
}

export default App;
