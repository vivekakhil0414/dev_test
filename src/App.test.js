import React from "react";
import App from "./App";

import renderer from "react-test-renderer";

test("First snapshot test", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
