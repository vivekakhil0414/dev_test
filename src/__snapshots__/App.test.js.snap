// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test 1`] = `
Array [
  <div
    className="App"
  >
    <div>
      <div
        className="sidenav"
      >
        <a
          href="/service"
          onClick={[Function]}
        >
          Show List
        </a>
      </div>
    </div>
    <div>
      Home Page
    </div>
  </div>,
  <div
    className="footer"
  >
    <p>
      All Copyrights Reserved
    </p>
  </div>,
]
`;
